File: app/src/main/java/com/sp/base/util/PermissionUtils.java
package com.sp.base.util;

import android.app.Activity;
import android.content.Context;
import android.content.pm.PackageManager;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import androidx.appcompat.app.AlertDialog;
import androidx.core.app.ActivityCompat;

public class PermissionUtils {
    public static int REQUEST_ID_MULTIPLE_PERMISSIONS = 1221;

    /* Check For Internet Connection Availability */
    public static boolean isInternetAvailable(Context context) {
        ConnectivityManager connectivity = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        if (connectivity != null) {
            NetworkInfo info = connectivity.getActiveNetworkInfo();
            if (info != null) {
                return info.getState() == NetworkInfo.State.CONNECTED || info.getState() == NetworkInfo.State.CONNECTING;
            }
        }
        return false;
    }

    // For Check multiple permissions
    public static boolean checkPermissions(Context context, String... permissions) {
        if (context != null && permissions != null) {
            for (String permission : permissions) {
                if (ActivityCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
                    return false;
                }
            }
        }
        return true;
    }

    public static void requestPermissions(Activity activity, String[] permissions) {
        if (!checkPermissions(activity.getBaseContext(), permissions)) {
            AlertDialog.Builder alertBuilder = new AlertDialog.Builder(activity);
            alertBuilder.setCancelable(true);
            alertBuilder.setTitle(("Permission"));
            alertBuilder.setMessage("Permission necessary");
            alertBuilder.setPositiveButton(android.R.string.yes, (dialog, which) -> {
                for (String permission : permissions) {
                    try {
//                        Timber.i(permission);
                        if (ActivityCompat.shouldShowRequestPermissionRationale(activity, permission)) {
                            ActivityCompat.requestPermissions(activity, permissions, REQUEST_ID_MULTIPLE_PERMISSIONS);
//                            Timber.i("permission denied, show dialog");
                        } else {
//                            Timber.i("else");
                            ActivityCompat.requestPermissions(activity, permissions, REQUEST_ID_MULTIPLE_PERMISSIONS);
                        }

                    } catch (IndexOutOfBoundsException e) {
                        e.printStackTrace();
                    }
                }
            });
            AlertDialog alert = alertBuilder.create();
            alert.show();
        }
    }
}

File: app/src/main/java/com/baseapplication/util/PermissionUtils.kt
package com.baseapplication.util

import android.app.Activity
import android.content.Context
import android.content.DialogInterface
import android.content.pm.PackageManager
import android.net.ConnectivityManager
import android.net.NetworkInfo
import androidx.appcompat.app.AlertDialog
import androidx.core.app.ActivityCompat

object PermissionUtils {
    var REQUEST_ID_MULTIPLE_PERMISSIONS: Int = 1221

    /* Check For Internet Connection Availability */
    fun isInternetAvailable(context: Context): Boolean {
        val connectivity: ConnectivityManager? =
            context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager?
        if (connectivity != null) {
            val info: NetworkInfo? = connectivity.getActiveNetworkInfo()
            if (info != null) {
                return info.getState() == NetworkInfo.State.CONNECTED || info.getState() == NetworkInfo.State.CONNECTING
            }
        }
        return false
    }

    // For Check multiple permissions
    fun checkPermissions(context: Context?, vararg permissions: String?): Boolean {
        if (context != null && permissions != null) {
            for (permission: String? in permissions) {
                if (ActivityCompat.checkSelfPermission(
                        context,
                        (permission)!!
                    ) != PackageManager.PERMISSION_GRANTED
                ) {
                    return false
                }
            }
        }
        return true
    }

    fun requestPermissions(activity: Activity, permissions: Array<String?>) {
        if (!checkPermissions(activity.getBaseContext(), *permissions)) {
            val alertBuilder: AlertDialog.Builder = AlertDialog.Builder(activity)
            alertBuilder.setCancelable(true)
            alertBuilder.setTitle(("Permission"))
            alertBuilder.setMessage("Permission necessary")
            alertBuilder.setPositiveButton(
                android.R.string.yes,
                DialogInterface.OnClickListener({ dialog: DialogInterface?, which: Int ->
                    for (permission: String? in permissions) {
                        try {
//                        Timber.i(permission);
                            if (ActivityCompat.shouldShowRequestPermissionRationale(
                                    activity,
                                    (permission)!!
                                )
                            ) {
                                ActivityCompat.requestPermissions(
                                    activity,
                                    permissions,
                                    REQUEST_ID_MULTIPLE_PERMISSIONS
                                )
                                //                            Timber.i("permission denied, show dialog");
                            } else {
//                            Timber.i("else");
                                ActivityCompat.requestPermissions(
                                    activity,
                                    permissions,
                                    REQUEST_ID_MULTIPLE_PERMISSIONS
                                )
                            }
                        } catch (e: IndexOutOfBoundsException) {
                            e.printStackTrace()
                        }
                    }
                })
            )
            val alert: AlertDialog = alertBuilder.create()
            alert.show()
        }
    }
}